### DISCUSSION TITLE
请问如何对组件进行二次开发？

### DISCUSSION THREAD

**Discussion by liaodalin19903** [ant-design/ant-design#44913] [Q&A]
现在有一个需求就是对Tree的item做点击事件，（取消select事件，而用点击事件）

![image](https://github.com/ant-design/ant-design/assets/12727582/049b0f44-67f1-4d04-a337-f543f1eed6af)

请问想要实现这样的功能，需要如何进行对组件进行二次开发呢？

**Comment by linxianxi** [comment]
可以 selectable={false} 配合 titleRender 加 onClick

**Reply by liaodalin19903** [reply]
非常感谢您的回复，但是现在有一个新的问题就是使用titleRender：
```
   <DirectoryTree
      titleRender={(nodeData) => {
        console.log('mmkk: ', nodeData)
        
        const nodeEle = (
          <span style={{display: 'inline', width: '100%'}} onClick={() => {
            console.log(1234)
          }}>{nodeData.title}</span>
        )

        // 如果拥有子组件
        if(nodeData.children) {
          return nodeData.title
        }
        return nodeEle 
      }}
```

我return的数据，只能在`<span class="ant-tree-title"></span>` 之内，我不能控制它占满一行：

![image](https://github.com/ant-design/ant-design/assets/12727582/72d775c2-0cc4-4a45-be75-41c2c5a054dc)

也就是说，我想要做点击事件让一行都可以点击上，但目前只能点击这个我返回的`nodeEle`一小部分。

**Reply by linxianxi** [reply]
blockNode 属性

**Reply by liaodalin19903** [reply]
谢谢老师的回复，但是还是做不到此需求。
我设置span为inline-block之后，
可能需要隐藏掉icon才能做到。
![image](https://github.com/ant-design/ant-design/assets/12727582/bf954a1f-a69d-4eff-a630-0fe7e6f944b1)

**Reply by linxianxi** [reply]
看不懂，提供一个 codesandbox 比较好

**Reply by liaodalin19903** [reply]
Codesandbox代码：
https://codesandbox.io/s/rg9h8d 

如果｛nodeData.title｝报错可以修改为任意字符串：test

**Reply by linxianxi** [reply]
设置 display: flex，然后 flex: 1 就行。然后 selectable 可以直接从组件传入，不用 data 里一个个写
